### PR TITLE
Adding two generic interfaces for creating any sentiment analysis model

### PR DESCRIPTION
**Pull request by MohamedElhawaty**

### CODE REVIEW

File: src/main/scala/edu/twitter/classification/Classifier.scala [hunk 1]
```diff
@@ -27,41 +26,21 @@ class Classifier(ssc: StreamingContext) {
     *
     * @return stream of `ClassifiedTweets`
     */
-  def createClassifiedStream(): DStream[ClassifiedTweet] = {
+  def createClassifiedStream(genericModelBuilder: GenericModelBuilder): DStream[ClassifiedTweet] = {
```
Comment (mohamedramadan2515) [inline src/main/scala/edu/twitter/classification/Classifier.scala]: Can you add parameters in the documentation?

File: src/main/scala/edu/twitter/model/TweetsLoader.scala [hunk 2]
```diff
@@ -29,10 +29,10 @@ class TweetsLoader(sc: SparkContext) {
     * @param messages All tweets text fields
     * @return tweets that contain happy word , tweets contains sad word
     */
-  private def cleanRecords(messages: DataFrame) : RDD[Row] = {
+  private def cleanRecords(messages: DataFrame): RDD[Row] = {
```
Comment (mohamedramadan2515) [inline src/main/scala/edu/twitter/model/TweetsLoader.scala]: We need to change this method doc

File: src/main/scala/edu/twitter/classification/Classifier.scala [hunk 3]
```diff
@@ -27,41 +26,21 @@ class Classifier(ssc: StreamingContext) {
...
     // Only checking for url patterns, mentions, hashtags and retweets
```
Comment (AbdulrahmanAttia) [inline src/main/scala/edu/twitter/classification/Classifier.scala]: I guess this comment should be moved to the  GradientBoostingModel class.

File: src/main/scala/edu/twitter/model/GradientBoostingModel.scala [hunk 4]
```diff
@@ -1,103 +1,23 @@
...
-    println("unhappy messages in " + setType + " Set: " + unhappyTotal + " happy messages: " + happyTotal)
-    println("happy % correct: " + happyCorrect.toDouble/happyTotal)
-    println("unhappy % correct: " + unhappyCorrect.toDouble/unhappyTotal)
-
-    var testErr = labelAndPreds.filter(r => r._1 != r._2).count.toDouble / data.count()
-    println("data size=" + data.count())
-    println("Test Error " + setType + " Set: " + testErr)
+  override def getLabel(tweetText: String): Double = {
+    val tokens = tweetText.split(" ").filter(!invalidTokens(_))
+    val features = hashingTF.transform(tweetText.split(" "))
```
Comment (AbdulrahmanAttia) [inline src/main/scala/edu/twitter/model/GradientBoostingModel.scala]: Shouldn't features be =  hashingTF.transform(tokens) ?

File: src/main/scala/edu/twitter/classification/Classifier.scala (hunk 1 above)
Comment (tarek-nawara) [reply src/main/scala/edu/twitter/classification/Classifier.scala]: Sure, will do it.

File: src/main/scala/edu/twitter/model/TweetsLoader.scala (hunk 2 above)
Comment (tarek-nawara) [reply src/main/scala/edu/twitter/model/TweetsLoader.scala]: Sure, will do it.

**Pull request merged**